### PR TITLE
Properly logs attrs

### PR DESCRIPTION
**Pull request by bartavelle**
Add missing fields, such as:

 * authority
 * method
 * content of mextra (items passed as meta that are not used by
   the filtering module)

### CODE REVIEW

**Review by flaper87 — DISMISSED** [review]

File: curiefense/curieproxy/rust/curiefense/src/utils.rs [hunk 1]
```diff
@@ -197,24 +197,37 @@ pub struct RequestInfo {
...
 impl RequestInfo {
     pub fn into_json(self, tags: Tags) -> serde_json::Value {
-        // TODO: geo
         let ipnum: Option<String> = self.rinfo.geoip.ip.as_ref().map(|i| match i {
             IpAddr::V4(a) => u32::from_be_bytes(a.octets()).to_string(),
             IpAddr::V6(a) => u128::from_be_bytes(a.octets()).to_string(),
         });
+        let geo = self.rinfo.geoip.to_json();
+        let mut attrs: HashMap<String, Option<String>> = [
+            ("uri", self.rinfo.qinfo.uri),
```
Comment (tzuryby) [inline curiefense/curieproxy/rust/curiefense/src/utils.rs:206]: Why don't you simply loop inside all entries within attrs?
Basically, every ":foo" should be there (in attr)

**Review by tzuryby — APPROVED** [review]
as suggested, looping through all attributes and metadata storing them for the log would be future compatible and avoid what is potentialy an error prone.

File: curiefense/curieproxy/rust/curiefense/src/utils.rs (hunk 1 above)
Comment (bartavelle) [reply curiefense/curieproxy/rust/curiefense/src/utils.rs:206]: I am not sure what you mean, as `attrs` is being built here, we can't loop into it.

Comment (bartavelle) [reply curiefense/curieproxy/rust/curiefense/src/utils.rs:206]: Ah I got it. Unrecognized arguments are stored in the mextra map, which is added to attrs, so this should be future proof.

File: curiefense/curieproxy/rust/curiefense/src/utils.rs [hunk 2]
```diff
@@ -197,24 +197,37 @@ pub struct RequestInfo {
...
+            ("query", Some(self.rinfo.qinfo.query)),
+            ("ip", Some(self.rinfo.geoip.ipstr)),
+            ("ipnum", ipnum),
+            ("authority", Some(self.rinfo.host)),
+            ("method", Some(self.rinfo.meta.method)),
+        ]
+        .iter()
+        .map(|(k, v)| (k.to_string(), v.clone()))
+        .collect();
+        attrs.extend(
```
Comment (bartavelle) [inline curiefense/curieproxy/rust/curiefense/src/utils.rs:217]: It is added here

**Review by tzuryby — APPROVED** [review]

**Pull request merged**